src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/IApplicationFeatureProviderOfT.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    /// <summary>
+    /// A provider for a given <typeparamref name="T"/> feature.
+    /// </summary>
+    /// <typeparam name="T">The type of the feature.</typeparam>
+    public interface IApplicationFeatureProvider<T> : IApplicationFeatureProvider
```

Comment:
Please remember to avoid using just `T`. Use something like `TFeature` instead. The only type of place where just `T` is used is when it has no particular meaning, which is almost never.